### PR TITLE
1.2.2 lambda-java-core release

### PR DESCRIPTION
**Pull request by andclt**
*Description of changes:*
### August 30, 2022
`1.2.2`:
- Added new `CustomPojoSerializer` interface
- Removed unnecessary usage of public on interface methods (aws#172) 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by smirnoal — APPROVED** [review]

**Comment by richarddd** [comment]
Wouldn't it also make sense to have a 4th method in the CustomSerializer:
`<T> String toJson(T value, Type type);`
Then the same serializer instance can be used inside handlers to create event responses requiring json strings

**Comment by andclt** [comment]
> Wouldn't it also make sense to have a 4th method in the CustomSerializer: `<T> String toJson(T value, Type type);` Then the same serializer instance can be used inside handlers to create event responses requiring json strings

As discussed offline, we do not want to force cx to implement that method when creating a custom serializer. Furthermore, if they want, they can always add it to their custom serializer class.

File: aws-lambda-java-core/src/main/java/com/amazonaws/services/lambda/runtime/CustomPojoSerializer.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+/**
+ * Interface required to implement a custom plain old java objects serializer
+ */
+public interface CustomPojoSerializer {
+
+    /**
+     * Deserializes from input stream to plain old java object
+     * @param input input stream
+     * @param type plain old java object type
+     * @return deserialized plain old java object of type T
```
Comment (pzygielo) [inline aws-lambda-java-core/src/main/java/com/amazonaws/services/lambda/runtime/CustomPojoSerializer.java:19]: What's `T`?

Comment (andclt) [reply aws-lambda-java-core/src/main/java/com/amazonaws/services/lambda/runtime/CustomPojoSerializer.java:19]: It's the type of the deserialized instance

Comment (pzygielo) [reply aws-lambda-java-core/src/main/java/com/amazonaws/services/lambda/runtime/CustomPojoSerializer.java:19]: I see the intention. But "deserialized plain old java object of type T" uses `T` which is undefined and unused anywhere else.

**Review by msailes — APPROVED** [review]
LGTM

**Pull request merged**